modules/cdc-ext/src/main/java/org/apache/ignite/cdc/kafka/IgniteToKafkaCdcStreamer.java

```diff
@@ -232,4 +242,139 @@ public IgniteToKafkaCdcStreamer(
     @Override public void stop() {
         producer.close();
     }
+
+    /** @return Whether entries only from primary nodes should be handled. */
+    public boolean isOnlyPrimary() {
+        return onlyPrimary;
+    }
+
+    /**
+     * Sets whether entries only from primary nodes should be handled.
+     *
+     * @param onlyPrimary Kafka whether entries only from primary nodes should be handled.
+     * @return {@code this} for chaining.
+     */
+    public IgniteToKafkaCdcStreamer setOnlyPrimary(boolean onlyPrimary) {
+        assert producer == null;
```

Comment:
Let's extract this and same checks into separate method like `ensureNotStart()`.

Let's have flag `private volatile boolean started`, also.
And set it to the `true` or `false` on start and stop.

Reply:
Second thought - do we have same check for `IgniteConfiguration`?
Do we really need one?

Reply:
Let's just remove this checks then. 
Done.